batch/batch/worker/worker.py

```diff
@@ -1171,6 +1189,7 @@ async def container_config(self):
                 'uidMappings': [],
                 'gidMappings': [],
                 'resources': {
+                    "devices": [{"allow": True, "type": "c", "major": 195, "minor": 0, "access": "rwm"}],
```

Comment:
Can you add a comment here on what this is doing or maybe a link to documentation?

Reply:
The documentation of the `linux.resource.devices` field is here: https://github.com/opencontainers/runtime-spec/blob/main/config-linux.md#devices.

Loosely, we need to specify in the config the (GPU) devices that we want to make available to the container.  Major mode 195 for Nvidia GPUs, see:

From https://www.kernel.org/doc/Documentation/admin-guide/devices.txt:

```
195 char	Nvidia graphics devices
		  0 = /dev/nvidia0		First Nvidia card
		  1 = /dev/nvidia1		Second Nvidia card
		    ...
		255 = /dev/nvidiactl		Nvidia card control device
```

However, there is still some behavior to the devices field we don't understand.  In particular, if we change this to device 5,0 /dev/tty, the container is still able to access the GPU!  (But if set `"devices": []` or omit the devices field, then the container cannot, as expected).

This appears to be related to cgroup device white/blacklists and default device permissions, which I'm still researching.